include/deal.II/matrix_free/cuda_hanging_nodes_internal.h

```diff
@@ -758,80 +758,82 @@ namespace CUDAWrappers
                 bool         transpose,
                 typename Number>
       __device__ inline void
-      interpolate_boundary_2d(const unsigned int constr, Number *values)
+      interpolate_boundary_2d(const unsigned int constraint_mask,
+                              Number *           values)
       {
         const unsigned int x_idx = threadIdx.x % (fe_degree + 1);
         const unsigned int y_idx = threadIdx.y;
 
         const unsigned int this_type =
           (direction == 0) ? internal::constr_type_x : internal::constr_type_y;
 
-        if (constr & (((direction == 0) ? internal::constr_face_y : 0) |
-                      ((direction == 1) ? internal::constr_face_x : 0)))
+        const unsigned int interp_idx = (direction == 0) ? x_idx : y_idx;
+
+        Number t = 0;
```

Comment:
What is this variable for?

Reply:
This is a variable that we use to compute the constrained value. It needs to be defined here because it is used in the two `if` that surround the `__synchtreads()`. Note that this is not a new variable, I have just moved it up.

Reply:
I know that it is not new, but I still don't understand what `t` is an abbreviation for. Anyway, not crucial to me.